library/src/main/java/com/phillipcalvin/iconbutton/IconButton.java

```diff
@@ -65,34 +66,47 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
         textPaint.getTextBounds(text, 0, text.length(), bounds);
 
         int textWidth = bounds.width();
-        int contentWidth = drawableWidth + iconPadding + textWidth;
+        int factor = (drawablePosition == DrawablePositions.LEFT_AND_RIGHT) ? 2 : 1;
+        int contentWidth = drawableWidth + iconPadding * factor + textWidth;
+        int horizontalPadding = (int) ((getWidth() / 2.0) - (contentWidth / 2.0));
+
+        setCompoundDrawablePadding(-horizontalPadding + iconPadding);
```

Comment:
That's correct, it's the space between the drawable and the text. By setting a negative value, we move them closer together. We move the drawable towards the text until it touches (`-horizontalPadding`) and then add back the space the user requested (`iconPadding`).

It's been a LONG TIME since I wrote this, but I think this is stolen from: <http://stackoverflow.com/questions/9968094/android-how-to-create-a-button-with-image-and-text-that-are-both-centred>. (The first ASCII diagram shows how things would look if we didn't set this value.)